rust/core/src/driver_manager.rs

```diff
@@ -316,6 +408,65 @@ impl Driver for ManagedDriver {
     }
 }
 
+fn get_optional_key(manifest: &Table, key: &str) -> String {
+    manifest
+        .get(key)
+        .and_then(Value::as_str)
+        .unwrap_or_default()
+        .to_string()
+}
+
+fn load_driver_manifest(info: &mut DriverInfo) -> Result<()> {
```

Comment:
I think it might be more idiomatic to take info by value and return it by value (rather than requiring a mutable reference)

Reply:
You can then write

```rust
let info = load_driver_manifest(DriverInfo { ... })?;
```